aten/src/ATen/ParallelNative.cpp

```diff
@@ -87,15 +86,15 @@ void _run_with_pool(const std::function<void(int, size_t)>& fn, size_t range) {
   // Run the first task on the current thread directly.
   fn(0, 0);
 #else
-  caffe2::ThreadPool* pool = caffe2::mobile_threadpool();
-  if (pool) {
-    // caffe2::ThreadPool can utilize the current thread.
-    pool->run(fn, range);
-  } else {
-    for (size_t i = 0; i < range; ++i) {
-      fn(0, i);
-    }
-  }
+  caffe2::PThreadPool* const pool = caffe2::pthreadpool();
+  TORCH_INTERNAL_ASSERT(pool, "Invalid thread pool!");
```

Comment:
We'll always have an implementation: the third-party version whenever that's available (i.e. platforms other than Windows), and C2's implementation if third-party is not available (which supports Windows.)

Reply:
So are we ensuring that on windows platform we are building C2's implementation?

Reply:
Third party pthreadpool shouldn't build on Windows because POSIX headers such as pthreads.h and unistd.h are not available there.  I haven't tested Windows locally, but I am taking Windows CIs passing to mean that at least something is not horribly broken. :)